YARG.Core/Engine/Vocals/VocalsEngine.cs

```diff
@@ -315,9 +322,22 @@ protected override void UpdateMultiplier()
 
         protected sealed override int CalculateBaseScore()
         {
-            return Notes.Where(note => note.ChildNotes.Count > 0).Sum(_ => EngineParameters.PointsPerPhrase);
+            return Notes.Where(note => note.IsLyricPhrase).Sum(_ => EngineParameters.PointsPerPhrase);
         }
 
         protected override bool CanSustainHold(VocalNote note) => throw new InvalidOperationException();
+
+        protected virtual void UpdateCarriedNote(VocalNote phrase)
+        {
+            if (CarriedVocalNote != null && CarriedVocalNote.TotalTickEnd > phrase.TickEnd)
+            {
+                // Keep the current Carried Vocal Note if it continues past the end of the current phrase.
+                // This can happen for notes that span across 3 or more phrases.
+                return;
+            }
+
+            CarriedVocalNote = phrase.ChildNotes.FirstOrDefault(e => !e.IsPercussion && e.TotalTickEnd > phrase.TickEnd);
```

Comment:
Again avoid using LINQ in engine code that runs repeatedly. This should be unfolded into an explicit loop:

```suggestion
            CarriedVocalNote = null;
            foreach (var note in phrase.ChildNotes)
            {
                if (e.IsPercussion && e.TotalTickEnd > phrase.TickEnd)
                {
                    CarriedVocalNote = note;
                    break;
                }
            }
```